From 3b55d694fb14cc25c19dcbf8fb67c075f9bc3f1e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Pierre-Cl=C3=A9ment=20Tosi?= <ptosi@google.com>
Date: Mon, 26 Sep 2022 08:56:57 +0000
Subject: [PATCH] libfdt: fdt_path_offset_namelen: Reject empty paths am:
 d10c84c4bc am: 9f4834b735

Original change: https://googleplex-android-review.googlesource.com/c/platform/external/dtc/+/20024811

Change-Id: I9af6b06514f4d63de88e801714b8acf7fb71c8ff
Signed-off-by: Automerger Merge Worker <android-build-automerger-merge-worker@system.gserviceaccount.com>
(cherry picked from commit e0440b09079afacb3101f5140560f719d438dadc)
Merged-In: I9af6b06514f4d63de88e801714b8acf7fb71c8ff
---
 libfdt/fdt_ro.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libfdt/fdt_ro.c b/libfdt/fdt_ro.c
index 08de2cc..3b65f16 100644
--- a/libfdt/fdt_ro.c
+++ b/libfdt/fdt_ro.c
@@ -188,6 +188,9 @@ int fdt_path_offset_namelen(const void *fdt, const char *path, int namelen)
 
 	FDT_CHECK_HEADER(fdt);
 
+	if (namelen < 1)
+		return -FDT_ERR_BADPATH;
+
 	/* see if we have an alias */
 	if (*path != '/') {
 		const char *q = memchr(path, '/', end - p);
-- 
2.38.1.273.g43a17bfeac-goog

